design/fips-compliance.md

```diff
@@ -0,0 +1,213 @@
+ authenticator-client, utilizing the `LOCAL_AUTHENTICATOR` environment variable.
+ If this is possible, we can trigger a `kubernetes-conjur-demo` build as part
+  of our build.
+
+This will give us full confidence in the green build as it also passed
+ integration tests.
+ 
+ However, the build time of `kubernetes-conjur-demo` is ~25 minutes (opposing
+  2-3 minutes of the `conjur-authn-k8s-client`) and it is not very stable
+  . This means that it will harder to merge PRs into the `conjur-authn-k8s
+  -client`.
+  
+I am not sure that this can be implemented. By setting`LOCAL_AUTHENTICATOR
+` to `true` we use the last built image (ran in the `./bin/build` step. In
+ case we have builds running in parallel it may affect the latest built image
+  so it is possible that we can't use a local authenticator in Jenkins.
+
+### Trigger `kubernetes-conjur-demo` to run after every master build of `conjur-authn-k8s-client`
+
+This option will still not fail `conjur-authn-k8s-client` builds that will
+fail in `kubernetes-conjur-demo` but we will get _some_ feedback. 
+After we will merge the PR into `master` we can follow the `master` build of 
+`kubernetes-conjur-demo` to see if it passed. This is better than a nightly
+build also because at the end of the day we will see which PR
+introduced a failure and it will be easier to fix the error. 
+  
+### Add integration tests to `conjur-authn-k8s-client` as we have in the `secrets-provider-for-kubernetes`
+  
+In the `secrets-provider-for-kubernetes` we have integration tests
+with different scenarios, that each one deploys its own pod and verifies
+the expected output.
+
+The caveat of this approach is that members of the community will not be able
+to contribute to this repo as they can today as they will not be able to run
+the integration tests (as they run with `summon` which requires `conjurops
+` access). They will not be able to fix tests that broke because of their
+change and will not be able to add tests for their contribution. However, this
+is not really different than today where community members cannot run the
+tests in `kubernetes-conjur-demo`, so it's better that tests will fail
+before we merge their PRs.
+
+### Summary
+
+To summarize the options, let's look at the following table:
+| **Option**                                                                             | **Build Time** | **Implementation time**                                                                                                                  | **Feedback on Failure**                                                           | **Ease of adding tests**                                                                                   | **Community Contribution**                                                                                                                                                                                                      |
```

Comment:
Two important issues related to this section:
1. Ease of adding tests - i don’t think its relevant , if its not easy we need to understand why and fix the problem
2.  kubernetes-conjur-demo is not stable - we have to make it stable anyway.
Those issues will take time but we need to see how we can improve both.

Reply:
Regarding section 2 - even if we will stabalize it, it is still more error prone as it has multiple environments.